otherlibs/systhreads/st_stubs.c

```diff
@@ -654,16 +654,11 @@ CAMLexport int caml_c_thread_register(void)
     return 0;
   }
   /* Add thread info block to the list of threads */
-  if (Active_thread == NULL) {
```

Comment:
The commit comment refers to `Thread_main_lock`. I can't find references to it. Has it been renamed to something else?

At a high-level, the invariant that we need is that `Active_thread` is non-NULL when `caml_c_thread_register` is called. The comment here says `Active_thread` may be NULL at safe point

https://github.com/ocaml/ocaml/blob/bbabb4b7075e49b5a5277633b5925e16c64f046e/otherlibs/systhreads/st_stubs.c#L138-L139

Is it not possible that `caml_c_thread_register` may be invoked at this point? 

(Meta comment: at this point, I have questions regarding the second commit, but I would be happy to have merged the first commit on its own.)

Reply:
Fixed `Thread_main_lock` -> `Thread_lock(Dom_c_threads)` in the commit description.

Roughly speaking, the `NULL` case for `Active_thread` is for before the threads are initialized on a domain, or during domain shutdown. But the code is not prepared to handle either case anyway, if it could ever happen. (Whether it should is a different question.)

Reply:
Sounds good.